staging/src/k8s.io/apiserver/pkg/server/storage/storage_factory.go

```diff
@@ -284,6 +289,20 @@ func (s *DefaultStorageFactory) NewConfig(groupResource schema.GroupResource) (*
 	}
 	glog.V(3).Infof("storing %v in %v, reading as %v from %#v", groupResource, codecConfig.StorageVersion, codecConfig.MemoryVersion, codecConfig.Config)
 
+	if storageConfig.Type == storagebackend.StorageTypeETCD3 {
```

Comment:
*really* not a fan of leaking the storage implementation here

Reply:
I can just always fill those details here. It's generic enough that it should be possible to implement in arbitrary backend.